src/main/scala/Tile.scala

```diff
@@ -0,0 +1,18 @@
+case class Tile(c: Char) {
+
+  def value: Int = {
```

Comment:
This is an interesting case! What if someone puts in a word like "groß"? Presumably it wouldn't be allowed because the tiles aren't in the bag, so it would fail further up in the validation, but it would be nice not to have to be concerned about that here, if we could just use local reasoning.

If, instead of matching a `Char`, we were to ensure that a `Tile` can only be one of A-Z, then here we would be able to have an exhaustive check and not worry about the `_` case. Better yet, we could fix the value of a given Tile/Letter so that we don't have to match at all:

```scala
sealed trait Tile {
    val value: Int
}
object Tile {
    case object A extends Tile { override val value = 1 }
    case object B extends Tile { override val value = 3 }
}
```